[WIP] Adding the Task Manager info to the UI doc

mpershina opened the pull request:
Adding the step showing the new Task Manager tab capabilities.

Tracked under [MTA-3110](https://issues.redhat.com/browse/MTA-3110).

The latest preview:

- [Analyzing an application](https://mpershina.github.io/Previews/MTA/MTA-3110-3#analyzing-an-application_user-interface-guide)
- [Configuring and running an application analysis](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-configuring-and-running-an-application-analysis_user-interface-guide)
- [Reviewing analysis details](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-reviewing-an-analysis-report_user-interface-guide)
- [Downloading an analysis report](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-downloading-an-analysis-report_user-interface-guide)
- [Controlling MTA tasks by using Task Manager](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-controlling-tasks-with-task-manager_user-interface-guide)
- [Reviewing a task log](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-reviewing-a-task-log_user-interface-guide)
- [Controlling the order of task execution](https://mpershina.github.io/Previews/MTA/MTA-3110-3#mta-web-controlling-task-order-with-task-manager_user-interface-guide)

mguetta1 commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,6 +92,17 @@ Analysis takes minutes to hours to run depending on the size of the application
 {ProductShortName} relies on Kubernetes scheduling capabilities to determine how many analyzer instances are created based on cluster capacity. If several applications are selected for analysis, by default, only one analyzer can be provisioned at a time. With more cluster capacity, more analysis processes can be executed in parallel.
 ====
+
+. Optional: To track the status of your active analysis tasks, click the *Task Manager* tab or the popover message on the application name. The *Task Manager* displays the following information:
```
Optional: To track the status of your active analysis tasks, open the *Task Manager drawer* by clicking the notification badge or the popover message on the application name

No need to mention the Task manager table here, IMO

mguetta1 commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,6 +92,21 @@ Analysis takes minutes to hours to run depending on the size of the application
++
+Alternatively, hover over the application name.
++
+The *Task Manager* displays the following information:
+
+** *ID*: The ID of the task.
+** *Application*: The application name associated with the task.
+** *Status*: The status of the task, for example, *Scheduled*, *Pending*, *In progress*, *Succeeded*, or *Failed*.
+** *Kind*: The type of the task, for example, *analyzer* or *discovery*.
+** *Priority*: Priority of the task.
+** *Preemption*: This column shows if the task must be executed prior to any other task. The values are *true* or *false*.
+** *Created By*: The name of the user who created the task.
+
```
@mpershina we are on the application inventory page at this step, please remove lines 100-108 which describe the table on the "Task Manager" page

cc: @dymurray @rromannissen to confirm

dymurray replied:
Agreed

dymurray commented:
I would like to see more details about how the user can actually interact with the task manager. This is also in the wrong section of the docs.

Could you provide more detail on how the user can navigate the task manager page?

mpershina commented:
> I would like to see more details about how the user can actually interact with the task manager. This is also in the wrong section of the docs.
> 
> Could you provide more detail on how the user can navigate the task manager page?

Thanks for your input, @dymurray! 

Hi @mguetta1, as per Dylan's input, the section is wrong.  So, to determine the section, I need to know when exactly would the user want to interact with the Task manager? We can also discuss which details the user would benefit from.

mpershina commented on docs/topics/mta-web-downloading-an-analysis-report.adoc:
```diff
@@ -6,6 +6,10 @@
 
+An {ProductShortName} analysis report contains a number of sections, including a listing of the technologies used by the application, the dependencies of the application, and the lines of code that must be changed to successfully migrate or modernize the application.
+
+For more information about the contents of an {ProductShortName} analysis report, see link:{ProductDocUserGuideURL}#review-reports_cli-guide[Reviewing the reports].
```
To be discussed whether this reference must be removed or not

mguetta1 replied:
Thanks @mpershina 
I just checked the static analysis report that can be downloaded from MTA UI and this reference looks relevant

mpershina replied:
Oki :-) Let's leave it as it is then.

mpershina commented:
@mguetta1, I attempted to provide some changes to the existing procedures and also created a new section as we discussed in our meeting. Could you please check? Let me know if you need me to send a preview of the doc :-)

Thank you!

mguetta1 commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
LGTM

mguetta1 commented on docs/topics/mta-web-reviewing-an-analysis-report.adoc:
LGTM

mguetta1 commented on docs/topics/mta-web-reviewing-a-task-log.adoc:
@mpershina could you please use information from @rromannissen's slides for this section?
https://docs.google.com/presentation/d/1_vXhNWLJaivlNvDCKkVFjV11VgUZVWiw0lKcZIMsGlk/edit#slide=id.g2f022e875fe_0_4

I will set us a meeting to explain which procedures should be documented. Thanks

mpershina replied:
Sure, thank you!

mguetta1 commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,7 +92,8 @@ Analysis takes minutes to hours to run depending on the size of the application
 ====
 {ProductShortName} relies on Kubernetes scheduling capabilities to determine how many analyzer instances are created based on cluster capacity. If several applications are selected for analysis, by default, only one analyzer can be provisioned at a time. With more cluster capacity, more analysis processes can be executed in parallel.
 ====
-. When analysis is complete, click the *Report* link to see the results of the analysis.
-. Optional: Display the details of the analysis:
-.. Click the Options menu ({kebab}).
-.. Select *Analysis details*. You can choose either the YAML or JSON format.
```
@mpershina This should be documented with changes. something like:
When analysis is complete, click the application name to open the application drawer and see the results of the analysis.

mpershina replied:
Attempted to rewrite it a bit, see the preview: [Configuring and running an application analysis](https://mpershina.github.io/Previews/MTA/MTA-3110#mta-web-configuring-and-running-an-application-analysis_user-interface-guide).

mpershina commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,7 +92,8 @@ Analysis takes minutes to hours to run depending on the size of the application
 ====
 {ProductShortName} relies on Kubernetes scheduling capabilities to determine how many analyzer instances are created based on cluster capacity. If several applications are selected for analysis, by default, only one analyzer can be provisioned at a time. With more cluster capacity, more analysis processes can be executed in parallel.
 ====
```
```suggestion
====
. When analysis is complete, to see its results, open the application drawer by clicking on the application name.
```

mpershina replied:
@mguetta1, will this work? :-) It's aimed to address your comment right below.

mpershina commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,7 +92,8 @@ Analysis takes minutes to hours to run depending on the size of the application
+. Optional: To track the status of your active analysis task, open the *Task Manager* drawer by clicking the notifications button.
++
+Alternatively, hover over the application name to display the pop-over window.
+. When analysis is complete, open the *Task Manager* page to display the analysis details.
```
```suggestion
. When analysis is complete, to see its results, open the application drawer by clicking on the application name.
+
Alternatively, open the *Task Manager* page to display the analysis details.
```

mpershina commented on docs/web-console-guide/master.adoc:
```diff
@@ -46,7 +46,7 @@ include::topics/mta-7-installing-web-console-on-openshift.adoc[leveloffset=+1]
 == Configuring the instance environment
 
-You can configure the following in *Administration* view:
+You can configure the followiby using Task Managerng in *Administration* view:
```
```suggestion
You can configure the following in *Administration* view:
```

mpershina commented on docs/web-console-guide/master.adoc:
```diff
@@ -77,7 +77,7 @@ include::topics/mta-web-config-source-control-credentials.adoc[leveloffset=+3]
 include::topics/mta-web-config-proxy-credentials.adoc[leveloffset=+3]
 
-[id="configuring-repositories"]
+[id="configuring-repositoriesby using Task Manager"]
```
```suggestion
[id="configuring-repositories"]
```

mpershina commented on docs/web-console-guide/master.adoc:
```diff
@@ -141,7 +141,7 @@ You can use the {ProductFullName} {WebName} to perform the following tasks:
 * Owner (optional, chosen from a list)
 * Contributors (optional, chosen from a list)
-* Source code (a path entered by the user)
+* Source code (a path enteredby using Task Manager by the user)
```
```suggestion
*  Source code (a path entered by the user)
```

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+.Procedure
+
+. In the {ProductShortName} user interface (UI), select the *Migration* view. 
+. Open the *Task Manager* page.
```
@mguetta1, it would be great to have a screenshot of how the Task Manager page looks like here :-) age.

mguetta1 replied:
Sure, I will send to you

mpershina replied:
Thanks! Received the screenshot - will attach it to the procedure.

mguetta1 commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,7 +92,14 @@ Analysis takes minutes to hours to run depending on the size of the application
+Alternatively, hover over the application name to display the pop-over window.
+. When analysis is complete, to see its results, open the application drawer by clicking on the application name.
++
+Alternatively, open the *Task Manager* page and perform either of the following steps:
+
+* Click on the status of the analysis task in the *Status* column.
+* Click the *Options* menu ({kebab}) for the analysis task and select *Task details*.
+
```
I recommend to remove this since that in the application drawer the user can access the entire report: analysis details, issues, dependencies, tags.
And via the Task manager page he can access the analysis details (activity log and attachments)

It will be mentioned here: `docs/topics/mta-web-controlling-tasks-with-task-manager.adoc`

mpershina replied:
Oki, removing then :-)

mguetta1 commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+[id="mta-web-controlling-task-order-with-task-manager_{context}"]
+= Controlling the order of task execution
+
+You can use *Task Manager* to preempt a {ProductFullName} task you have scheduled for execution to rearrange execution priority. 
```
@mpershina I think that `{ProductFullName}` is redundant here, but it is up to you.

mpershina replied:
It's just our doc rules. We should mention full names when the the utility/technology appears in the section for the first time.

mguetta1 replied:
I see. I meant to not mention here the project name at all

mpershina replied:
Yeah, could be an option, of course, I guess, it's just better for the reader :-) Let's see what the doc reviewer will say about it.

mguetta1 commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+You can only set *Preemption* for the following tasks:
+
+* Automated tasks (language discovery or technology discovery)
+* Task you scheduled yourself (application analysis). These tasks have priority over automated tasks.
```
Please remove `These tasks have priority over automated tasks.` - I don't think it is always true.

mpershina replied:
Oki, got it! I just took that from the presentation.

mguetta1 commented on docs/topics/mta-web-controlling-tasks-with-task-manager.adoc:
Please use this section for sub-titles:
1. `docs/topics/mta-web-controlling-task-order-with-task-manager.adoc`
2. `docs/topics/mta-web-reviewing-a-task-log.adoc`

mpershina replied:
If I understood correctly, this two sections as subsections for [Controlling MTA tasks by using Task Manager](https://mpershina.github.io/Previews/MTA/MTA-3110-1#mta-web-controlling-tasks-with-task-manager_user-interface-guide).

mguetta1 replied:
Correct

mpershina replied:
Then everything should be ok with the preview now :-)

mguetta1 commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+. In the {ProductShortName} user interface (UI), select the *Migration* view. 
+. Open the *Task Manager* page.
+. Click the *Options* menu ({kebab}) for the selected task.
+. Depending on you your scenario, complete one of the following steps:
```
Please remove redundant word `you`

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+You can only set *Preemption* for the following tasks:
+
+* Automated tasks (language discovery or technology discovery)
+* Task you scheduled yourself (application analysis). These tasks have priority over automated tasks.
```
```suggestion
* Task you scheduled yourself (application analysis).
```

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,26 @@
+. In the {ProductShortName} user interface (UI), select the *Migration* view. 
+. Open the *Task Manager* page.
+. Click the *Options* menu ({kebab}) for the selected task.
+. Depending on you your scenario, complete one of the following steps:
```
```suggestion
. Depending on your scenario, complete one of the following steps:
```

mpershina commented on docs/topics/mta-web-configuring-and-running-an-application-analysis.adoc:
```diff
@@ -92,7 +92,14 @@ Analysis takes minutes to hours to run depending on the size of the application
++
+Alternatively, hover over the application name to display the pop-over window.
+. When analysis is complete, to see its results, open the application drawer by clicking on the application name.
++
+Alternatively, open the *Task Manager* page and perform either of the following steps:
+
+* Click on the status of the analysis task in the *Status* column.
+* Click the *Options* menu ({kebab}) for the analysis task and select *Task details*.
```
```suggestion
```

mguetta1 replied:
LGTM

mpershina replied:
Thanks!

mpershina commented:
@mguetta1, I added the screenshot (not visible in the html preview, unfortunately) to the procedure and applied our suggestions. Please, could you check this PR once more as the final QE review? If you have no further suggestions, I'll send it to Dylan :-)

sjd78 commented on docs/topics/mta-web-controlling-tasks-with-task-manager.adoc:
```diff
@@ -0,0 +1,14 @@
+* Language discovery
+* Technology discovery
+
+You can display task-related information by using either the *Task Manager* drawer or the *Task Manager* page in the *Migration* view.
```
```suggestion
You can display task-related information by using either the *Task Manager* drawer for active tasks, or the *Task Manager* page in the *Migration* view for all tasks.
```

sjd78 commented on docs/topics/mta-web-reviewing-an-analysis-report.adoc:
```diff
@@ -4,16 +4,16 @@
-. Click *Reports*.
-. Click the dependencies or source links.
-. Click the tabs to review the report.
+. Click on the application raw to open the application drawer. 
```
```suggestion
. Click on the application row to open the application drawer. 
```

sjd78 commented on docs/topics/mta-web-reviewing-an-analysis-report.adoc:
```diff
@@ -4,16 +4,16 @@
 :_content-type: PROCEDURE
 [id="mta-web-reviewing-an-analysis-report_{context}"]
-= Reviewing an analysis report
+= Reviewing analysis details
```
It is my understanding that the analysis details is not the same as the old analysis report.

@jortel, @dymurray - In older versions, was there a link to open the windup report from the application details drawer?  Is that what the older docs are referencing?

If that is the case, the analysis report is now just either downloading the static report or the Issues and Dependencies from Details tab of the application details drawer.

sjd78 reviewed (changes requested):
Most of it looks reasonable.  I have a few suggestions.

mguetta1 commented on docs/topics/mta-web-controlling-tasks-with-task-manager.adoc:
```diff
@@ -0,0 +1,14 @@
+* Language discovery
+* Technology discovery
+
+You can display task-related information by using either the *Task Manager* drawer for active tasks, or the *Task Manager* page in the *Migration* view for all tasks.
```
@mpershina Please mention how to open the drawer "by clicking the notifications badge/button"

mpershina replied:
Done, @mguetta1, see my suggestion below :-)

mguetta1 replied:
LGTM

mguetta1 commented on docs/topics/mta-web-reviewing-a-task-log.adoc:
```diff
@@ -0,0 +1,17 @@
+
+.Procedure
+
+. In the *Migration* view, click *Application inventory*.
```
@mpershina the user does not need to click on `Application inventory` menu,  but: "In the **Migration** view, click **Task Manager**"

mpershina replied:
Eh, my bad, copy-paste error :-D

mguetta1 commented on docs/topics/mta-web-downloading-an-analysis-report.adoc:
@mpershina could you please accurate steps 4 and 5 to make it similar to the previous sub-section:
```
Click on the application row to open the application drawer.
Click the Reports tab.
```

mguetta1 commented on docs/topics/mta-web-reviewing-an-analysis-report.adoc:
```diff
@@ -4,16 +4,16 @@
-. Click *Reports*.
-. Click the dependencies or source links.
-. Click the tabs to review the report.
+. Click on the application row to open the application drawer. 
```
@mpershina could you please check the preview page? there is a typo there:
"Click on the application **raw** to open the application drawer."

mpershina replied:
The preview is being manually generated, so the current preview is just a commit or two behind the actual code. Will update the preview links.

mpershina commented on docs/topics/mta-web-controlling-tasks-with-task-manager.adoc:
```diff
@@ -0,0 +1,14 @@
+* Language discovery
+* Technology discovery
+
+You can display task-related information by using either the *Task Manager* drawer for active tasks, or the *Task Manager* page in the *Migration* view for all tasks.
```
```suggestion
You can display task-related information either of the following ways:

* To display active tasks, open the *Task Manager* drawer by clicking the notifications button.
* To display all tasks, open the *Task Manager* page in the *Migration* view.
```

mpershina commented on docs/topics/mta-web-reviewing-a-task-log.adoc:
```diff
@@ -0,0 +1,17 @@
+
+.Procedure
+
+. In the *Migration* view, click *Application inventory*.
```
```suggestion
. In the *Migration* view, click *Task Manager*.
```

mpershina commented:
Hi @mguetta1, thanks for your suggestions! I incorporated your feedback and updated the preview links above. Is there anything else I should fix for this PR or is it ready in your opinion?
Thanks!

mguetta1 commented on docs/topics/mta-web-reviewing-a-task-log.adoc:
```diff
@@ -0,0 +1,17 @@
+.Procedure
+
+. In the *Migration* view, click *Task Manager*.
+. Open the *Task Manager* page.
```
@mpershina please remove step 2

mguetta1 commented on docs/topics/mta-web-reviewing-a-task-log.adoc:
```diff
@@ -0,0 +1,17 @@
+
+. In the *Migration* view, click *Task Manager*.
+. Open the *Task Manager* page.
+. Click the *Options* menu ({kebab}).
```
@mpershina please mention that this is for a task row
`Click the Options menu ( kebab ) for the selected task. ` (copied from next sub-section)

mguetta1 commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,29 @@
+. In the {ProductShortName} user interface (UI), select the *Migration* view. 
+. Open the *Task Manager* page.
++
+image::task-manager-page.png[Task Manager]
```
I suggest that the image will be somewhere in the beginning of the task manager section:
`docs/topics/mta-web-controlling-tasks-with-task-manager.adoc`

mguetta1 reviewed (approved):
LGTM. Thanks!

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,25 @@
+[id="mta-web-controlling-task-order-with-task-manager_{context}"]
+= Controlling the order of task execution
+
+You can use *Task Manager* to preempt a {ProductFullName} task you have scheduled for execution to rearrange execution priority. 
```
```suggestion
You can use *Task Manager* to preempt a {ProductFullName} task you have scheduled for execution. 
```

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,25 @@
+
+[NOTE]
+====
+You can only set *Preemption* for the following tasks:
+
+* Automated tasks (language discovery or technology discovery)
+* Task you scheduled yourself (application analysis).
```
```suggestion
You can only set *Preemption* for automated tasks (language discovery or technology discovery).
```

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,25 @@
+
+. In the *Migration* view, click *Task Manager*.
+. Click the *Options* menu ({kebab}) for the selected task.
+. Depending on your scenario, complete one of the following steps:
+
+** To enable *Preemption* for the task, select *Enable preemption*. 
+** To disable *Preemption* for the task with enabled *Preemption*, select *Disable preemption*.
```
```suggestion
. Enable *Preemption* for the task by selecting *Enable preemption*. 
```

jortel commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,22 @@
+[NOTE]
+====
+You can only set *Preemption* for automated tasks (language discovery or technology discovery).
+====
```
You can enable preemption on any task.
However, it's only has an affect on higher priority tasks such as analysis.  Thus, only useful to enable on analysis tasks.

mpershina replied:
@mguetta1, please, check the new text we created together with @jortel.

jortel replied:
The 2nd.
> lower-priority tasks are candidates to be preempted means that they are candidates to be rescheduled in favor of higher tasks?

mpershina commented on docs/topics/mta-web-controlling-task-order-with-task-manager.adoc:
```diff
@@ -0,0 +1,22 @@
+
+[NOTE]
+====
+You can only set *Preemption* for automated tasks (language discovery or technology discovery).
```
```suggestion
You can enable *Preemption* on any scheduled task (not in the status of *Running*, *Succeeded*, or *Failed*). However, only lower-priority tasks are candidates to be preempted. When a higher-priority task is blocked by lower-priority tasks and has *Preemption* enabled, the low-priority tasks might be rescheduled so that the blocked higher-priority task might run. Therefore, it is only useful to enable *Preemption* on higher-priority tasks, for example, application analysis.
```
